Development to review

ppolyanskiyy opened the pull request:
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)


## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

ppolyanskiyy committed:
Added new RealTimeFaceMaskDetector solution, added Windows Service project

ppolyanskiyy committed:
Update .gitignore

DemienBlogan committed:
Add files via upload

DemienBlogan committed:
Add files via upload

DemienBlogan committed:
Add files via upload

DemienBlogan committed:
Add files via upload

DemienBlogan committed:
Add files via upload

DemienBlogan committed:
Delete TutorialScript.sql

DemienBlogan committed:
Delete TestCode.cpp

DemienBlogan committed:
Delete README.txt

DemienBlogan committed:
Delete pch.h

DemienBlogan committed:
Delete pch.cpp

DemienBlogan committed:
Delete framework.h

DemienBlogan committed:
Delete dllmain.cpp

DemienBlogan committed:
Delete SQLServer.h

DemienBlogan committed:
Delete SQLServer.cpp

DemienBlogan committed:
Delete SQLServer.dll

DemienBlogan committed:
Delete SQLServer.exp

DemienBlogan committed:
Delete SQLServer.iobj

DemienBlogan committed:
Delete SQLServer.ipdb

DemienBlogan committed:
Delete SQLServer.lib

DemienBlogan committed:
Delete SQLServer.pdb

DemienBlogan committed:
Delete SQLServer.dll

DemienBlogan committed:
Delete SQLServer.exp

DemienBlogan committed:
Delete SQLServer.ilk

DemienBlogan committed:
Delete SQLServer.lib

DemienBlogan committed:
Delete SQLServer.pdb

rprokopiv committed:
EncryptDecrypt

rprokopiv committed:
added cryptlib.lib

rprokopiv committed:
Updated settings

rprokopiv committed:
Added .gitattributes

ziolushka committed:
Add Server part

rprokopiv committed:
deleted cryptlib.lib

rprokopiv committed:
deleted lfs from repository

DmytroAntoniuk committed:
added opencv library and face recognition component

Sapik26 committed:
Add TCP Client

DemienBlogan committed:
Adding .DLL project

Stepbura commented on RealTimeFaceMaskDetector/SQLServerDLL/SQLServer.h:
```diff
@@ -0,0 +1,39 @@
+#include <vector>
+#include <SQLAPI.h>
+
+class SQLSERVER_API SQLServer
```
It would better to create an interface for your class and implementation

rprokopiv committed:
added asymmetric encrypt/decrypt

Sapik26 committed:
Merged Client and EncryptDecrypt

marko-yaciv committed:
Ppolyanskiyy to dev (#4)

* Full project added

* Update Configuration.conf

* Update Logger.h

* Delete Configuration.conf

* Update Logger.h

* Update Configuration.conf

* Add files via upload

* Update Configuration.conf

* Delete Configuration.conf

Co-authored-by: Andrew Polyanskiy <ppolyanskiyy@gmail.com>
Co-authored-by: Andrew Polyanskiy <71780594+ppolyanskiyy@users.noreply.github.com>

OleksiiiSemko committed:
Deleted folder 3rdPartyLibraries

ChyzhTolik committed:
Lvc 231 (#8)

* Full project added

* Update Configuration.conf

* Update Logger.h

* Delete Configuration.conf

* Update Logger.h

* Update Configuration.conf

* Add files via upload

* Update Configuration.conf

* Delete Configuration.conf

* Full project added

* Update Configuration.conf

* Update Logger.h

* Delete Configuration.conf

* Update Logger.h

* Update Configuration.conf

* Add files via upload

* Update Configuration.conf

* Delete Configuration.conf

* It works O_o

* Refactored SQLServer

* Chyzh (#1)

* Full project added

* Update Configuration.conf

* Update Logger.h

* Delete Configuration.conf

* Update Logger.h

* Update Configuration.conf

* Add files via upload

* Update Configuration.conf

* Delete Configuration.conf

* It works O_o

* Ppolyanskiyy to dev (#4)

* Full project added

* Update Configuration.conf

* Update Logger.h

* Delete Configuration.conf

* Update Logger.h

* Update Configuration.conf

* Add files via upload

* Update Configuration.conf

* Delete Configuration.conf

Co-authored-by: Andrew Polyanskiy <ppolyanskiyy@gmail.com>
Co-authored-by: Andrew Polyanskiy <71780594+ppolyanskiyy@users.noreply.github.com>

* Full project added

* Update Configuration.conf

* Update Logger.h

* Delete Configuration.conf

* Update Logger.h

* Update Configuration.conf

* Add files via upload

* Update Configuration.conf

* Delete Configuration.conf

* It works O_o

* Refactored SQLServer

Co-authored-by: Andrew Polyanskiy <ppolyanskiyy@gmail.com>
Co-authored-by: Andrew Polyanskiy <71780594+ppolyanskiyy@users.noreply.github.com>
Co-authored-by: MarkiianYatsiv <61264740+marko-yaciv@users.noreply.github.com>

* added config.ini and CreatePhotos files. They need to be placed in the application dir

* Added files config and CreateTable

* Corrected CreateTable

* rm config

* Full project added

* Update Configuration.conf

* Update Logger.h

* Update Logger.h

* Update Configuration.conf

* Add files via upload

* Update Configuration.conf

* Delete Configuration.conf

* Full project added

* Update Logger.h

* Delete Configuration.conf

* Update Logger.h

* Update Configuration.conf

* Add files via upload

* Update Configuration.conf

* Delete Configuration.conf

* added config.ini and CreatePhotos files. They need to be placed in the application dir

* Added files config and CreateTable

* Corrected CreateTable

* rm config

* Added unit test for IniParser

* Added unit test

* Deleted TestFile

* curly brackets

Co-authored-by: Andrew Polyanskiy <ppolyanskiyy@gmail.com>
Co-authored-by: Andrew Polyanskiy <71780594+ppolyanskiyy@users.noreply.github.com>
Co-authored-by: MarkiianYatsiv <61264740+marko-yaciv@users.noreply.github.com>

OleksiiiSemko committed:
-Added script to download dependencies
-Changed links in vs project

OleksiiiSemko committed:
Moved folder 3rdPartylibs on one level up

OleksiiiSemko committed:
- Edited headers
- Edited vs config

OleksiiiSemko committed:
Changed qt path

OleksiiiSemko committed:
Included direcories in the config

OleksiiiSemko committed:
add new include path

OleksiiiSemko committed:
added platform to TCPClient

OleksiiiSemko committed:
added path envs in conanfile

OleksiiiSemko committed:
linked opencv lib

OleksiiiSemko committed:
added build dependencies

OleksiiiSemko committed:
edited solution config

sotnyk-lv committed:
Remove absolute path in SQLServerDLL vcxproj. (#11)

Remove redundant search of face on binary image.
Change search of the mouth and nose from the whole image to the
corresponding face.

marko-yaciv committed:
LVC-221 (#2) (#10)

* Full project added

* Update Configuration.conf

* Update Logger.h

* Delete Configuration.conf

* Update Logger.h

* Update Configuration.conf

* Add files via upload

* Update Configuration.conf

* Delete Configuration.conf

* Configured and builded project with installed opencv and prebuilded  cryptocpp820 in 3rdPartyLibs folder.
To run server sqlapi.dll file should be in the same folder as .exe files. To run TCPClient file opecv_world450d.dll should be in the same folder as .exe files.

* added ability fo SocketServer to receive photos all time

* Server now creates folder with photos and writes there every photo it received

* Fxed bug with receiving correct photo. Now every photo is received in one recv call. Added exception when receive function fails.

* Listening is executed in the other thread. Also when client has connected to server, Receiving starts in the other thread

* both merged code works properly. Photos are being written in local database and saved in folder

Co-authored-by: Andrew Polyanskiy <ppolyanskiyy@gmail.com>
Co-authored-by: Andrew Polyanskiy <71780594+ppolyanskiyy@users.noreply.github.com>

ChyzhTolik committed:
Logger into lib (#14)

* Full project added

* Update Configuration.conf

* Update Logger.h

* Delete Configuration.conf

* Update Logger.h

* Update Configuration.conf

* Add files via upload

* Update Configuration.conf

* Delete Configuration.conf

* It works O_o

* Full project added

* Update Configuration.conf

* Update Logger.h

* Delete Configuration.conf

* Update Logger.h

* Update Configuration.conf

* Add files via upload

* Update Configuration.conf

* Delete Configuration.conf

* It works O_o

* Refactored SQLServer

* Chyzh (#1)

* Full project added

* Update Configuration.conf

* Update Logger.h

* Delete Configuration.conf

* Update Logger.h

* Update Configuration.conf

* Add files via upload

* Update Configuration.conf

* Delete Configuration.conf

* It works O_o

* Ppolyanskiyy to dev (#4)

* Full project added

* Update Configuration.conf

* Update Logger.h

* Delete Configuration.conf

* Update Logger.h

* Update Configuration.conf

* Add files via upload

* Update Configuration.conf

* Delete Configuration.conf

Co-authored-by: Andrew Polyanskiy <ppolyanskiyy@gmail.com>
Co-authored-by: Andrew Polyanskiy <71780594+ppolyanskiyy@users.noreply.github.com>

* Full project added

* Update Configuration.conf

* Update Logger.h

* Delete Configuration.conf

* Update Logger.h

* Update Configuration.conf

* Add files via upload

* Update Configuration.conf

* Delete Configuration.conf

* It works O_o

* Refactored SQLServer

Co-authored-by: Andrew Polyanskiy <ppolyanskiyy@gmail.com>
Co-authored-by: Andrew Polyanskiy <71780594+ppolyanskiyy@users.noreply.github.com>
Co-authored-by: MarkiianYatsiv <61264740+marko-yaciv@users.noreply.github.com>

* added config.ini and CreatePhotos files. They need to be placed in the application dir

* Added files config and CreateTable

* Corrected CreateTable

* rm config

* Full project added

* Update Configuration.conf

* Update Logger.h

* Update Logger.h

* Update Configuration.conf

* Add files via upload

* Update Configuration.conf

* Delete Configuration.conf

* Full project added

* Update Logger.h

* Delete Configuration.conf

* Update Logger.h

* Update Configuration.conf

* Add files via upload

* Update Configuration.conf

* Delete Configuration.conf

* added config.ini and CreatePhotos files. They need to be placed in the application dir

* Added files config and CreateTable

* Corrected CreateTable

* rm config

* Added unit test for IniParser

* Added unit test

* Deleted TestFile

* Lvc 231 (#2)

* Full project added

* Update Configuration.conf

* Update Logger.h

* Delete Configuration.conf

* Update Logger.h

* Update Configuration.conf

* Add files via upload

* Update Configuration.conf

* Delete Configuration.conf

* It works O_o

* Ppolyanskiyy to dev (#4)

* Full project added

* Update Configuration.conf

* Update Logger.h

* Delete Configuration.conf

* Update Logger.h

* Update Configuration.conf

* Add files via upload

* Update Configuration.conf

* Delete Configuration.conf

Co-authored-by: Andrew Polyanskiy <ppolyanskiyy@gmail.com>
Co-authored-by: Andrew Polyanskiy <71780594+ppolyanskiyy@users.noreply.github.com>

* Full project added

* Update Configuration.conf

* Update Logger.h

* Delete Configuration.conf

* Update Logger.h

* Update Configuration.conf

* Add files via upload

* Update Configuration.conf

* Delete Configuration.conf

* It works O_o

* Refactored SQLServer

* added config.ini and CreatePhotos files. They need to be placed in the application dir

* Added files config and CreateTable

* Corrected CreateTable

* rm config

* Full project added

* Update Configuration.conf

* Update Logger.h

* Update Logger.h

* Update Configuration.conf

* Add files via upload

* Update Configuration.conf

* Delete Configuration.conf

* Full project added

* Update Logger.h

* Delete Configuration.conf

* Update Logger.h

* Update Configuration.conf

* Add files via upload

* Update Configuration.conf

* Delete Configuration.conf

* added config.ini and CreatePhotos files. They need to be placed in the application dir

* Added files config and CreateTable

* Corrected CreateTable

* rm config

* Added unit test for IniParser

* Added unit test

* Deleted TestFile

Co-authored-by: Andrew Polyanskiy <ppolyanskiyy@gmail.com>
Co-authored-by: Andrew Polyanskiy <71780594+ppolyanskiyy@users.noreply.github.com>
Co-authored-by: MarkiianYatsiv <61264740+marko-yaciv@users.noreply.github.com>

* curly brackets

* New unit test

* Added Logger Lib

* Changed Table Photos

* Added Logger Lib

* write date to database

* Fixed table create

* fixed insert

Co-authored-by: Andrew Polyanskiy <ppolyanskiyy@gmail.com>
Co-authored-by: Andrew Polyanskiy <71780594+ppolyanskiyy@users.noreply.github.com>
Co-authored-by: MarkiianYatsiv <61264740+marko-yaciv@users.noreply.github.com>

OleksiiiSemko committed:
resolved conflicts and setup gtest library

OleksiiiSemko committed:
edited config

OleksiiiSemko committed:
Task/lv 271 (#18)

* Deleted folder 3rdPartyLibraries

* -Added script to download dependencies
-Changed links in vs project

* Moved folder 3rdPartylibs on one level up

* - Edited headers
- Edited vs config

* Changed qt path

* Included direcories in the config

* add new include path

* added platform to TCPClient

* added path envs in conanfile

* linked opencv lib

* added build dependencies

* edited solution config

* edited config

OleksiiiSemko committed:
added dependencies

ChyzhTolik committed:
Move connect to server (#24)

* Full project added

* Update Configuration.conf

* Update Logger.h

* Delete Configuration.conf

* Update Logger.h

* Update Configuration.conf

* Add files via upload

* Update Configuration.conf

* Delete Configuration.conf

* It works O_o

* Full project added

* Update Configuration.conf

* Update Logger.h

* Delete Configuration.conf

* Update Logger.h

* Update Configuration.conf

* Add files via upload

* Update Configuration.conf

* Delete Configuration.conf

* It works O_o

* Refactored SQLServer

* Chyzh (#1)

* Full project added

* Update Configuration.conf

* Update Logger.h

* Delete Configuration.conf

* Update Logger.h

* Update Configuration.conf

* Add files via upload

* Update Configuration.conf

* Delete Configuration.conf

* It works O_o

* Ppolyanskiyy to dev (#4)

* Full project added

* Update Configuration.conf

* Update Logger.h

* Delete Configuration.conf

* Update Logger.h

* Update Configuration.conf

* Add files via upload

* Update Configuration.conf

* Delete Configuration.conf

Co-authored-by: Andrew Polyanskiy <ppolyanskiyy@gmail.com>
Co-authored-by: Andrew Polyanskiy <71780594+ppolyanskiyy@users.noreply.github.com>

* Full project added

* Update Configuration.conf

* Update Logger.h

* Delete Configuration.conf

* Update Logger.h

* Update Configuration.conf

* Add files via upload

* Update Configuration.conf

* Delete Configuration.conf

* It works O_o

* Refactored SQLServer

Co-authored-by: Andrew Polyanskiy <ppolyanskiyy@gmail.com>
Co-authored-by: Andrew Polyanskiy <71780594+ppolyanskiyy@users.noreply.github.com>
Co-authored-by: MarkiianYatsiv <61264740+marko-yaciv@users.noreply.github.com>

* added config.ini and CreatePhotos files. They need to be placed in the application dir

* Added files config and CreateTable

* Corrected CreateTable

* rm config

* Full project added

* Update Configuration.conf

* Update Logger.h

* Update Logger.h

* Update Configuration.conf

* Add files via upload

* Update Configuration.conf

* Delete Configuration.conf

* Full project added

* Update Logger.h

* Delete Configuration.conf

* Update Logger.h

* Update Configuration.conf

* Add files via upload

* Update Configuration.conf

* Delete Configuration.conf

* added config.ini and CreatePhotos files. They need to be placed in the application dir

* Added files config and CreateTable

* Corrected CreateTable

* rm config

* Added unit test for IniParser

* Added unit test

* Deleted TestFile

* Lvc 231 (#2)

* Full project added

* Update Configuration.conf

* Update Logger.h

* Delete Configuration.conf

* Update Logger.h

* Update Configuration.conf

* Add files via upload

* Update Configuration.conf

* Delete Configuration.conf

* It works O_o

* Ppolyanskiyy to dev (#4)

* Full project added

* Update Configuration.conf

* Update Logger.h

* Delete Configuration.conf

* Update Logger.h

* Update Configuration.conf

* Add files via upload

* Update Configuration.conf

* Delete Configuration.conf

Co-authored-by: Andrew Polyanskiy <ppolyanskiyy@gmail.com>
Co-authored-by: Andrew Polyanskiy <71780594+ppolyanskiyy@users.noreply.github.com>

* Full project added

* Update Configuration.conf

* Update Logger.h

* Delete Configuration.conf

* Update Logger.h

* Update Configuration.conf

* Add files via upload

* Update Configuration.conf

* Delete Configuration.conf

* It works O_o

* Refactored SQLServer

* added config.ini and CreatePhotos files. They need to be placed in the application dir

* Added files config and CreateTable

* Corrected CreateTable

* rm config

* Full project added

* Update Configuration.conf

* Update Logger.h

* Update Logger.h

* Update Configuration.conf

* Add files via upload

* Update Configuration.conf

* Delete Configuration.conf

* Full project added

* Update Logger.h

* Delete Configuration.conf

* Update Logger.h

* Update Configuration.conf

* Add files via upload

* Update Configuration.conf

* Delete Configuration.conf

* added config.ini and CreatePhotos files. They need to be placed in the application dir

* Added files config and CreateTable

* Corrected CreateTable

* rm config

* Added unit test for IniParser

* Added unit test

* Deleted TestFile

Co-authored-by: Andrew Polyanskiy <ppolyanskiyy@gmail.com>
Co-authored-by: Andrew Polyanskiy <71780594+ppolyanskiyy@users.noreply.github.com>
Co-authored-by: MarkiianYatsiv <61264740+marko-yaciv@users.noreply.github.com>

* curly brackets

* New unit test

* Added Logger Lib

* Changed Table Photos

* Added Logger Lib

* write date to database

* Fixed table create

* fixed insert

* Moved Connect

* Move Connect

Co-authored-by: Andrew Polyanskiy <ppolyanskiyy@gmail.com>
Co-authored-by: Andrew Polyanskiy <71780594+ppolyanskiyy@users.noreply.github.com>
Co-authored-by: MarkiianYatsiv <61264740+marko-yaciv@users.noreply.github.com>

OleksiiiSemko committed:
fixed merge conflicts

OleksiiiSemko committed:
fixed merge conflicts

OleksiiiSemko committed:
fixed conflicts

OleksiiiSemko committed:
Added installation steps

OleksiiiSemko committed:
Merge pull request #28 from OleksiiiSemko/hotfix/lvc-296

Hotfix/lvc 296

OleksiiiSemko committed:
Merge pull request #30 from OleksiiiSemko/task/LVC-275

Task/lvc 275

ShostakOleh committed:
Empty photos (#35)

* Fixed empty photos bug

* Added threadsafe recv

* Deleta SocketServer empty tests

* Delete empty server tests

Co-authored-by: Anatolii Chyzh <76211432+ChyzhTolik@users.noreply.github.com>
